### PR TITLE
Restart Numbering Command

### PR DESCRIPTION
**Pull request by BDeveau**
# What it does - issue #13 
Adds a button to restart numbering on H1 elements

- command is disabled for anything other than H1
- on if current H1 is set to restart numbering
- toggles autonumber-restart class
- CSS class added to enable the restarting
- tests set up to test the various states and classes

### CODE REVIEW

File: structuredheadings/styles/numbering.css [hunk 1]
```diff
@@ -26,6 +26,10 @@ h6.autonumber {
 	counter-increment: h6;
 }
 
+h1.autonumber-restart {
+	counter-reset: h1;
+}
```
Comment (caffodian) [inline structuredheadings/styles/numbering.css:31]: Should this be h1.autonumber.autonumber-restart ?

Comment (caffodian) [reply structuredheadings/styles/numbering.css:31]: I also wonder a bit about the virtues of this over having some separate element that restarts things (like a section break.)  I haven't given it much thought though.

File: structuredheadings/plugin.js [hunk 2]
```diff
@@ -1,10 +1,11 @@
-/*eslint max-statements: [2, 25]*/ //I'll need to refactor somehow to meet this rule
+/*eslint max-statements: [2, 50]*/ //I'll need to refactor somehow to meet this rule
```
Comment (caffodian) [inline structuredheadings/plugin.js:1]: oh dear.  50

File: structuredheadings/plugin.js [hunk 3]
```diff
@@ -1,10 +1,11 @@
...
 (function () {
   CKEDITOR.plugins.add("structuredheadings", {
     icons: "autonumberheading," +
            "matchheading," +
            "increaseheadinglevel," +
-           "decreaseheadinglevel",
+           "decreaseheadinglevel," +
+           "restartNumbering",
```
Comment (caffodian) [inline structuredheadings/plugin.js:8]: alternately,
```
var commandNames = [
'matchheading', 
'increaseheadinglevel',
....
'restartNumbering',]
...
icons: commandNames.join(','),
```

**Review by caffodian — APPROVED** [review]
looks good, looks like you've hit a good groove for progress!

File: structuredheadings/styles/numbering.css (hunk 1 above)
Comment (BDeveau) [reply structuredheadings/styles/numbering.css:31]: Hm... I'm adding autonumber and autonumber-restart at the same time. I'm not sure how much it matters... As far as the separate element, I'm just following Spec right now :)

File: structuredheadings/plugin.js (hunk 2 above)
Comment (BDeveau) [reply structuredheadings/plugin.js:1]: Yep, tired of it :)
BUT... I have spent some time looking around at the way some other plugins are structured (MOST seem to throw everything under init just like I am) and I think I maybe found another way of doing things, I'm going to try it.

File: structuredheadings/plugin.js (hunk 3 above)
Comment (BDeveau) [reply structuredheadings/plugin.js:8]: The only thing, is that simplifies the icons but I still need to list them all out individually somewhere, and that commandNames list would only be used for the icons anyway. This way is kinda cutting out the middleman.

**Pull request merged**